posix/System/File/Platform.hs

```diff
@@ -37,7 +38,7 @@ openExistingFile fp iomode = fdToHandle_ iomode fp =<< case iomode of
   df = defaultFileFlags { noctty = True, nonBlock = True, creat = Nothing }
 
 fdToHandle_ :: IOMode -> PosixPath -> Fd -> IO Handle
-fdToHandle_ iomode fp (Fd fd) = do
+fdToHandle_ iomode fp (Fd fd) = (`onException` closeFd (Fd fd)) $ do
```

Comment:
from what I understand, `fdToHandle'` does not close the fd on failure, so I guess we need the `onException` here?